src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,24 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private static final int MIN_AGE = 35;
+    private static final int INCREMENT_NUMBER = 1;
+    private static final int MIN_PERIOD = 10;
+
+    @Override
+    public boolean test(Candidate candidate) {
+        return candidate.getAge() >= MIN_AGE
+                && candidate.isAllowedToVote()
+                && candidate.getNationality().equals("Ukrainian")
+                && getPeriodsInUkraine(candidate.getPeriodsInUkr());
+    }
+
+    private boolean getPeriodsInUkraine(String period) {
```

Comment:
```suggestion
    private int getPeriodsInUkraine(String period) {
```
would be more logical if you return number of years in ukraine form this method
move all bussines logic checks to the test method